packages/twenty-front/src/modules/window-event/components/WindowEventProvider.tsx

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useRecoilValue, useSetRecoilState } from 'recoil';
+
+import { tokenPairState } from '@/auth/states/tokenPairState';
+import { chromeExtensionIdState } from '@/client-config/states/chromeExtensionIdState';
+
+const isInFrame = () => {
+  try {
+    return window.self !== window.top;
+  } catch (e) {
+    return true;
+  }
+};
+
+export const WindowEventProvider: React.FC<React.PropsWithChildren> = ({
+  children,
+}) => {
+  const navigate = useNavigate();
+  const setTokenPair = useSetRecoilState(tokenPairState);
+  const chromeExtensionId = useRecoilValue(chromeExtensionIdState);
+  const [isLoadingTokens, setIsLoadingTokens] = useState(true);
+
+  useEffect(() => {
+    if (isInFrame()) {
+      window.parent.postMessage(
+        'loaded',
+        `chrome-extension://${chromeExtensionId}`,
+      );
+
+      const handleWindowEvents = (event: MessageEvent<any>) => {
+        if (event.origin === `chrome-extension://${chromeExtensionId}`) {
+          switch (event.data.type) {
+            case 'tokens': {
+              setTokenPair(event.data.value);
+              setIsLoadingTokens(false);
+              break;
+            }
+            case 'navigate':
+              navigate(event.data.value);
+              break;
+            default:
+              break;
+          }
+        }
+      };
+      window.addEventListener('message', handleWindowEvents);
+      return () => {
+        window.removeEventListener('message', handleWindowEvents);
+      };
+    }
+  }, [chromeExtensionId, setIsLoadingTokens, setTokenPair, navigate]);
+
+  return isInFrame() ? !isLoadingTokens && <>{children}</> : <>{children}</>;
```

Comment:
Would be nice to have a dedicated error message in case you're in Frame but not the chrome extension to avoid crazy debugging where people don't realize it's the reason it fails

Reply:
Please provide clear error message in any scenario by the way. I realize `isLoadingTokens` is always false for me but took me 10 minutes to realize it was the issue (?)

Reply:
Not sure what I should do?
<img width="872" alt="Screenshot 2024-05-28 at 09 53 59" src="https://github.com/twentyhq/twenty/assets/6399865/47516966-ec19-4e2f-a549-90f4594e8bc6">